pkg/resource/apply.go

```diff
@@ -66,7 +66,15 @@ func (r *helper) Apply(obj []byte) (ApplyResult, error) {
 	// fakeApply mode (used in scale testing only) doesn't run the apply. Instead it will Sleep for a period of time
 	// that matches our request time metrics in real hive environments to simulate i/o wait.
 	fakeAppliesEnvVar, _ := os.LookupEnv(constants.HiveSyncSetsFakeApplyEnvVar)
-	if fakeApplies, err := strconv.ParseBool(fakeAppliesEnvVar); fakeApplies && err == nil {
+	if fakeApplies, err := strconv.ParseBool(fakeAppliesEnvVar); r.fake || (fakeApplies && err == nil) {
```

Comment:
> (fakeApplies && err == nil)

it seems like we are accessing fakeApplies output before checking for err here, imo other way around would be better?

Reply:
Code now gone due to the fakeHelper refactor.